Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -865,7 +822,17 @@ private bool RegisterServiceInternal(Type interfaceType, IMixedRealityService se
         private bool RegisterServiceInternal<T>(T serviceInstance) where T : IMixedRealityService
```

Comment:
>RegisterServiceInternal [](start = 21, length = 23)

Can we have some comments about why only this flavor of RegisterServiceInternal is also calling MixedRealityServiceRegistry.AddService<T>(serviceInstance, this)?

For example, there are comments saying that core services are always removable but I see them get added like:


#region Services Initialization

            var orderedCoreSystems = activeSystems.OrderBy(m => m.Value.Priority).ToArray();
            activeSystems.Clear();

            foreach (var system in orderedCoreSystems)
            {
                RegisterServiceInternal(system.Key, system.Value);
            }

            var orderedServices = registeredMixedRealityServices.OrderBy(service => service.Item2.Priority).ToArray();
            registeredMixedRealityServices.Clear();

            foreach (var service in orderedServices)
            {
                RegisterServiceInternal(service.Item1, service.Item2);
            }

            InitializeAllServices();


And these wouldn't get added to the registry because they don't call this overload of RegisterServiceInternal.

Reply:
you bet!

Reply:
added comment as recommended